### DISCUSSION TITLE
Question: how could I export everything from local into the docker

### DISCUSSION THREAD

**Discussion by lanyusea** [apache/superset#18293] [Q&A / Help]
I installed the superset by pip and there has been a lot of data: users/chars/dashboards.

now I docker-compose up the latest docker image in order to use some unreleased features, but I'm wondering how could I export all existing data from my local filesystem into the contianer.

there is the .superset/superset.db in my home folder from the one installed from `pip`, but I cannot find one inside the running containers to take the place of. also the `superset db` has no export/import related commands.

Is there any way that I can move my existing data into the docker image?

**Comment by PoLabs** [comment]
Also interested in this, except I need to move my docker superset dashboards to another machine

**Comment by sam-hieken** [comment]
Hey @lanyusea , did you ever end up finding a solution to this? Also experiencing the same issue.

I ended up trying to mount my old database directly into the container (e.g. `docker run -v ~/.superset:/app/superset_home ... apache/superset`) which did let me log in with my old credentials, but led to errors when trying to open my old dashboards. I'm trying to upgrade between superset versions though which may have caused this, so could be worth a try if you're not (just keep a backup if you do).

**Comment by mistersister** [comment]
Late, but for reference, this as always worked for me:

A) Just dump your metadata db. regular pg_dump/mysql_dump etc should suffice. Then import into your docker database. Before importing, stop all docker containers other that the database (to prevent them to create connections to the db) delete the superset database, create it empty and load your dump. Then, enter the superset_app container to run the docker upgrade db command to update the sql from the database and superset db init to apply changes.

This has always worked for me when trying upgrade versions that required different postgres versions 14 to 15 for instance.
